src/main/java/apoc/coll/Coll.java

```diff
@@ -127,6 +127,33 @@ public long indexOf(@Name("coll") List<Object> coll, @Name("value") Object value
     }
 
     @UserFunction
+    @Description("apoc.coll.set(coll, index, value) | set index to value")
```

Comment:
should we be more lenient with  these functions

* on coll null -> return null
* on invalid indexes (<0 or > length == null ) ?  -> just return the original coll unchanged
* and then please test for these cases

* can you add a function like `insertList` that inserts a list ?


The only unfortunate drawback is that these functions won't keep the list-type, so one also will have to convert them if a specific type is needed.

Reply:
@jexp yes i should make these more lenient. your suggestions look good. also yes to tests. i'm hoping to get to this soon.

what do you mean these won't keep the list-type? the only type supported by cypher is list type. cypher doesn't support Set so i don't see why these wouldn't always be a List type.

Reply:
Yes but you might pass in a List<Node> but what these functions will return is a List<Object>, so you loose the compound type `Node`